chalk-ir/src/cast.rs

```diff
@@ -193,23 +193,6 @@ impl<I: Interner> CastTo<GenericArg<I>> for GenericArg<I> {
     }
 }
 
-impl<T, I> CastTo<ProgramClause<I>> for T
```

Comment:
...but maybe we should keep this impl, which would let us revert the diffs below.

Reply:
I didn't check this one, whoops.